File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/tea/AmountKindTest.java
package coolpharaoh.tee.speicher.tea.timer.core.tea;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class AmountKindTest {

    @Test
    public void getTextGram() {
        final AmountKind amountKind = AmountKind.GRAM;
        assertThat(amountKind.getText()).isEqualTo("Gr");
    }

    @Test
    public void getChoiceGram() {
        final AmountKind amountKind = AmountKind.TEA_SPOON;
        assertThat(amountKind.getChoice()).isZero();
    }

    @Test
    public void amountKindFromTextGram() {
        final AmountKind amountKind = AmountKind.fromText("Gr");
        assertThat(amountKind).isEqualTo(AmountKind.GRAM);
    }

    @Test
    public void amountKindFromTextNotDefined() {
        final AmountKind amountKind = AmountKind.fromText("not defined");
        assertThat(amountKind).isEqualTo(AmountKind.TEA_SPOON);
    }

    @Test
    public void amountKindFromTextNull() {
        final AmountKind amountKind = AmountKind.fromText(null);
        assertThat(amountKind).isEqualTo(AmountKind.TEA_SPOON);
    }

    @Test
    public void amountKindFromChoiceZero() {
        final AmountKind amountKind = AmountKind.fromChoice(1);
        assertThat(amountKind).isEqualTo(AmountKind.GRAM);
    }

    @Test
    public void amountKindFromChoiceMinusOne() {
        final AmountKind amountKind = AmountKind.fromChoice(-1);
        assertThat(amountKind).isEqualTo(AmountKind.TEA_SPOON);
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/tea/AmountKindTest.kt
package coolpharaoh.tee.speicher.tea.timer.core.tea

import coolpharaoh.tee.speicher.tea.timer.core.tea.AmountKind.Companion.fromChoice
import coolpharaoh.tee.speicher.tea.timer.core.tea.AmountKind.Companion.fromText
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class AmountKindTest {
    @Test
    fun getTextGram() {
        val amountKind = AmountKind.GRAM
        assertThat(amountKind.text).isEqualTo("Gr")
    }

    @Test
    fun choiceGram() {
        val amountKind = AmountKind.TEA_SPOON
        assertThat(amountKind.choice).isZero
    }

    @Test
    fun amountKindFromTextGram() {
        val amountKind = fromText("Gr")
        assertThat(amountKind).isEqualTo(AmountKind.GRAM)
    }

    @Test
    fun amountKindFromTextNotDefined() {
        val amountKind = fromText("not defined")
        assertThat(amountKind).isEqualTo(AmountKind.TEA_SPOON)
    }

    @Test
    fun amountKindFromTextNull() {
        val amountKind = fromText(null)
        assertThat(amountKind).isEqualTo(AmountKind.TEA_SPOON)
    }

    @Test
    fun amountKindFromChoiceZero() {
        val amountKind = fromChoice(1)
        assertThat(amountKind).isEqualTo(AmountKind.GRAM)
    }

    @Test
    fun amountKindFromChoiceMinusOne() {
        val amountKind = fromChoice(-1)
        assertThat(amountKind).isEqualTo(AmountKind.TEA_SPOON)
    }
}